### PR TITLE
User list

### PR DESCRIPTION
**Pull request by FrankIglesias**
## Summary

- User list

### CODE REVIEW

File: app/services/user.js [hunk 1]
```diff
@@ -12,6 +12,16 @@ exports.findByEmail = email =>
...
   });
 
+exports.getUsers = ({ page = 0, limit = 20 }) =>
+  User.findAll({ offset: page * limit, limit }).then(users => {
+    return new Promise(resolve => {
+      resolve({
+        page: users,
+        current_page: page + 1
+      });
+    });
```
Comment (arinaldi118) [inline app/services/user.js]: A service’s responsibility it’s quite trivial: it should interact with external services and database integration. We should keep it as simple as possible. Here we are a document with best practices https://docs.google.com/document/d/1QOmG2yvZg0IhUwzx-lnOruvMEmUJLNx7bnk_SZ6cFJU/edit#heading=h.wn6nf5zepult (in progress jajajaja)

File: app/middlewares/auth.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const jwt = require('jwt-simple'),
+  errors = require('../errors'),
+  User = require('../services/user');
+
+exports.validCredentials = (req, res, next) => {
+  const token = req.headers.authorization;
+  if (!token) next(errors.unauthorizedError);
+  const { email, expirationDate } = jwt.decode(token, '123');
+  if (new Date() > new Date(expirationDate)) next(errors.expiredTokenError);
```
Comment (arinaldi118) [inline app/middlewares/auth.js]: you must use `moment`

File: app/middlewares/auth.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  User.findByEmail(email).then(user => {
+    if (!user) next(errors.unauthorizedError);
```
Comment (arinaldi118) [inline app/middlewares/auth.js]: added return

File: app/middlewares/auth.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    req.user = user;
+    next();
```
Comment (arinaldi118) [inline app/middlewares/auth.js]: Idem

File: app/controllers/user.js [hunk 5]
```diff
@@ -38,7 +38,19 @@ exports.signIn = (req, res, next) => {
...
+    })
+    .catch(next);
+};
+
+exports.getUsers = (req, res, next) => {
+  User.getUsers(req.params)
+    .then(users => {
+      res.status(200).send({
+        page: users,
+        current_page: req.params.page + 1
```
Comment (arinaldi118) [inline app/controllers/user.js]: if you don't set page in the body, page will be undefined, and undefined + 1 = nan. BOOM!!!! jajajaja

Maybe you should add a query builder

File: app/services/user.js [hunk 6]
```diff
@@ -12,6 +12,12 @@ exports.findByEmail = email =>
     throw errors.databaseError(err.detail);
   });
 
+exports.getUsers = ({ page = 0, limit = 20 }) =>
+  User.findAll({ offset: page * limit, limit }).carch(error => {
```
Comment (arinaldi118) [inline app/services/user.js]: idem

File: app/controllers/user.js [hunk 7]
```diff
@@ -38,7 +38,19 @@ exports.signIn = (req, res, next) => {
...
+        res.status(200).send({ user, token });
+      });
+    })
+    .catch(next);
+};
+
+exports.getUsers = (req, res, next) => {
+  User.getUsers(req.params)
+    .then(users => {
+      res.status(200).send({
```
Comment (arinaldi118) [inline app/controllers/user.js]: Maybe you can add a serializer or a mapper to build the response

File: app/services/user.js (hunk 6 above)
Comment (FrankIglesias) [reply app/services/user.js]: check the default param I added to the function. Though I think a query builder is not a bad choice

File: app/services/jwtService.js [hunk 8]
```diff
@@ -15,3 +15,6 @@ exports.comparePasswords = (password, user) =>
     );
     return { user, token };
   });
+
+exports.createToken = (email, creationDate = moment()) =>
+  jwt.encode({ email, creationDate, expirationDate: creationDate.add('days', 2) }, '123');
```
Comment (arinaldi118) [inline app/services/jwtService.js]: Que villero :exito: en algun momento mandale una constante al menos jajajaja

**Pull request merged**